platforms/android/tangram/src/main/cpp/androidPlatform.cpp

```diff
@@ -355,20 +361,21 @@ void AndroidPlatform::onUrlComplete(JNIEnv* _jniEnv, jlong _jRequestHandle, jbyt
         response.error = error.c_str();
     }
 
-    // Find the callback associated with the request.
-    UrlCallback callback;
-    {
-        std::lock_guard<std::mutex> lock(m_callbackMutex);
-        UrlRequestHandle requestHandle = static_cast<UrlRequestHandle>(_jRequestHandle);
-        auto it = m_callbacks.find(requestHandle);
-        if (it != m_callbacks.end()) {
-            callback = std::move(it->second);
-            m_callbacks.erase(it);
+
+    m_fileWorker.enqueue([this, _jRequestHandle, r = std::move(response)]() mutable {
```

Comment:
What's the purpose of sending this to another thread? Moving heavy callbacks off of the networking threads? A comment explaining this would be helpful since it might not always be obvious.

Reply:
> Don't let http requests block reading local files: Since there is a fixed amount of UrlClient workers all of them can be busy waiting for network responses. This blocks reading local files until network calls completed.

Reply:
Whoops wrong section..
Yeah it's done for moving decoding of scenes images from Okhttp threads.